Packs/MongoDBAtlas/Integrations/MongoDBAtlas/MongoDBAtlas.py

```diff
@@ -0,0 +1,586 @@
+    if min_date:
+        results = client.get_events_request(min_date=min_date)
+    else:  # first time fetching events
+        results, new_min_time = first_time_fetching_events(client, fetch_limit)
+        new_last_run_obj = create_last_run_dict_for_events(results, new_min_time)
+        return results, new_last_run_obj
+
+    response = get_last_page_of_events(client, results)
+    links = response.get('links', [])
+    events = response.get('results', [])
+
+    current_fetched_events_amount = 0
+    output = []
+    new_min_date = min_date
+
+    while current_fetched_events_amount < fetch_limit:
+        for event in reversed(events):  # ignore
+            event_id = event.get('id')
+            if event_id in events_with_created_min_time:
+                continue
+
+            enrich_event(event, 'events')
+            output.append(event)
+            current_fetched_events_amount += 1
+            new_min_date = get_latest_date(new_min_date, event.get('created'))
+
+            demisto.debug(f'Fetched event ID {event_id} from type "events"')
+
+            if current_fetched_events_amount == fetch_limit:
+                demisto.debug(f'Fetch limit reached. Total events fetched: {len(output)}')
+                new_last_run_obj = create_last_run_dict_for_events(output, new_min_date)
+                return output, new_last_run_obj
+
+        previous_page = get_page_url(links, page_type='previous')
+        if previous_page:
+            response = client.get_response_from_page_link(previous_page)
+            events = response.get('results', [])
+            links = response.get('links', [])
+        else:
+            break
+
+    demisto.debug(f'No more events left to fetch. Total events fetched: {len(output)}')
+    new_last_run_obj = create_last_run_dict_for_events(output, new_min_date)
+    return output, new_last_run_obj
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """
+    Returning 'ok' indicates that the integration works like it suppose to. Connection to the service is successful.
+
+    Args:
+        client: MongoDB Atlas client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test
+    """
+    client.get_alerts_request(page_num=1, items_per_page=1)
+    return 'ok'
+
+
+def fetch_events(client: Client, fetch_limit: int):
+    last_run = demisto.getLastRun()
+
+    alerts_output, last_run_alerts = fetch_alert_type(client, fetch_limit, last_run)
+    events_output, last_run_events = fetch_event_type(client, fetch_limit, last_run)
+
+    last_run_new_obj = {**last_run_alerts, **last_run_events}
+    return (alerts_output + events_output), last_run_new_obj
+
+
+def get_events(client: Client, args):
+    fetch_limit = int(args.get('limit'))
+
+    last_run = demisto.getLastRun()
+
+    alerts_output, _ = fetch_alert_type(client, fetch_limit, last_run)
+    events_output, _ = fetch_event_type(client, fetch_limit, last_run)
```

Comment:
Why are you not using `fetch_events` here?
Also, where are you referring here to `should_push_events`?

Reply:
should_push_event is checked in main